relay-server/src/actors/store.rs

```diff
@@ -1092,7 +1092,8 @@ impl Message for KafkaMessage {
             Self::Profile(_message) => Uuid::nil(),
             Self::ReplayEvent(message) => message.replay_id.0,
             Self::ReplayRecordingNotChunked(_message) => Uuid::nil(), // Ensure random partitioning.
-            Self::CheckIn(_message) => Uuid::nil(),
+            // Group associated check-ins into the same partition for better ordering.
+            Self::CheckIn(message) => message.check_in_id.0,
```

Comment:
If I understand this correctly, the `check_in_id` is unique for every check-in, while the monitor slug identifies the monitor? If that's the case, this strategy would not group anything together.

https://github.com/getsentry/relay/blob/8aa5ecde5261c2a522a42f37ba83d493a1827d00/relay-monitors/src/lib.rs#L127-L133

Reply:
> the check_in_id is unique for every check-in,

This is not actually entirely true, it is possible, and many SDKs do this, to send the SAME check-in for 'updating' check-ins.

What I am talking about si when you have check ins that look like this

```
- 01:00:23 -- 69054ffa-16df-459a-8b3d-6282fb75f973: in_progress
- 01:00:28 -- 69054ffa-16df-459a-8b3d-6282fb75f973: ok (5 second run)
- 01:01:20 -- f0d4caca-354e-49b4-894b-41a842bd9ff0: in_progress
- 01:00:30 -- f0d4caca-354e-49b4-894b-41a842bd9ff0: ok (10 second run)
```

It is not strictly required that you send the check-in ID, in fact, for HTTP style check-ins using the endpoints on relay, check-ins may have a nil ID

```
- 01:00:23 -- 00000000-0000-0000-0000-000000000000: in_progress
- 01:00:28 -- 00000000-0000-0000-0000-000000000000: ok (5 second run)
- 01:01:20 -- 00000000-0000-0000-0000-000000000000: in_progress
- 01:00:30 -- 00000000-0000-0000-0000-000000000000: ok (10 second run)
```

What happens in this scenario is that in our consumer we will identify "the most recent" check-in that is not in a terminal state (`ok`, `error`, `timed_out` are terminal state. `in_progress` is not)

---

What this change is fixing is **the first scenario**. We want those `in_progress` and `ok` check-ins to be in the same partition.

In the future ideally what we'll do is handle ordering issues across partitions at the consumer level. But for now this quick change can hopefully improve ordering reliability.

Reply:
> it is possible, and many SDKs do this, to send the SAME check-in for 'updating' check-ins

Thanks, good to know! Might be worth adding this information to the doc comment above `check_in_id`.